test/integration/backward_compatible/parallel/serialization/compact/CompactPublicAPIsTest.js

```diff
@@ -0,0 +1,874 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const chai = require('chai');
+const Long = require('long');
+const sinon = require('sinon');
+const sandbox = sinon.createSandbox();
+chai.use(require('chai-as-promised'));
+const should = chai.should();
+
+const RC = require('../../../../RC');
+const TestUtil = require('../../../../../TestUtil');
+const CompactUtil = require('./CompactUtil');
+const { Predicates } = require('../../../../../../lib');
+
+class Car {
```

Comment:
Is there a reason we have different class? I mean, would Inner or Outer suffice?